Yoast/wordpress-seo
# Set the Schema HowTo name and Article headline to the post title with a fallback

## Summary


This PR can be summarized in the following changelog entry:

* Set the Schema HowTo name and Article headline to the post title with a fallback to "No title"

## Relevant technical choices:

- introduces `WPSEO_Schema_Utils::get_post_title_with_fallback()`
- adds a test for a HowTo within a post with no title
- I took the liberty of fixing also the Article `headline` in the case where the post title is empty, as it originally used `get_the_title()` which is used internally by the new `WPSEO_Schema_Utils::get_post_title_with_fallback()`

## Test instructions

This PR can be tested by following these steps:

- create a post with a title, a How-to block, and publish
- check the Schema output on the front end

**On trunk:**
- see the HowTo `name` is set to whatever the SEO title is set to in the snippet editor
- for example, in the second page of a paginated post this value:
`%%title%% %%sep%% %%page%% %%sep%% %%primary_category%% %%sep%% %%sitename%% `
would produce:
`"name": "My title - Page 2 of 2 - MyCategory - WordPress stable",`
- on a standard post, the default value `%%title%% %%page%% %%sep%% %%sitename%% `
would produce:
`"name": "My awesome post title - WordPress stable",`
- see the Article `headline` is set to the post title, for example: `headline": "My awesome post title",`
- edit the post and remove the title so that it has an empty title, them save
- check the Schema output
- the HowTo name is: `"name": "- Page 2 of 2 - MyCategory - WordPress stable",`
- the Article headline is empty: `"headline": "",`

**On this branch:**
- repeat the steps above
- with a post title:
  - HowTo name: `"name": "My awesome post title",`
  - Article headline: `"headline": "My awesome post title",`
- with empty post title:
  - HowTo name: `"name": "No title",`
  - Article headline: `"headline": "No title",`


## UI changes

## Documentation

## Quality assurance


Fixes https://github.com/Yoast/wordpress-seo/issues/13259
